### PR TITLE
Simulartor hang oom

### PR DESCRIPTION
**Pull request by yhwen**
When shutting down the client, calls the logout to the server, so the server will be able to notify all the running jobs that this client has been dead. When the controller detects all the participating clients are dead, this will fire the system_panic to end the job run.

### CODE REVIEW

**Review by yanchengnv — CHANGES_REQUESTED** [review]

File: nvflare/private/fed/client/fed_client_base.py [hunk 1]
```diff
@@ -263,6 +263,17 @@ def send_heartbeat(self, project_name):
         except FLCommunicationError:
             self.communicator.heartbeat_done = True
 
+    def quite_remote(self, project_name, fl_ctx: FLContext):
```
[the remark this answers is not in the document]
Comment (yhwen) [reply nvflare/private/fed/client/fed_client_base.py]: a type. Fixed.

File: nvflare/private/fed/client/fed_client_base.py [hunk 2]
```diff
@@ -341,17 +352,22 @@ def start_heartbeat(self):
...
             fl_ctx: FLContext
 
         Returns: N/A
 
         """
-        return self.communicator.quit_remote(self.servers, task_name, self.token, self.ssid, fl_ctx)
+        pool = None
+        try:
+            pool = ThreadPool(len(self.servers))
+            return pool.map(partial(self.quite_remote, fl_ctx=fl_ctx), tuple(self.servers))
```
[the remark this answers is not in the document]
Comment (yhwen) [reply nvflare/private/fed/client/fed_client_base.py]: Fixed.

File: nvflare/private/fed/client/fed_client_base.py [hunk 3]
```diff
@@ -341,17 +352,22 @@ def start_heartbeat(self):
...
             fl_ctx: FLContext
 
         Returns: N/A
 
         """
-        return self.communicator.quit_remote(self.servers, task_name, self.token, self.ssid, fl_ctx)
+        pool = None
+        try:
+            pool = ThreadPool(len(self.servers))
+            return pool.map(partial(self.quit_remote, fl_ctx=fl_ctx), tuple(self.servers))
```
[the remark this answers is not in the document]
Comment (yhwen) [reply nvflare/private/fed/client/fed_client_base.py:367]: One FL server can support multiple FL projects at the same time. It uses the project_name to distinguish which service. This is consistent with all the other service functions.

**Pull request merged**